services/galley/src/Galley/API/LegalHold.hs

```diff
@@ -333,27 +342,88 @@ disableForUser zusr tid uid (Public.DisableLegalHoldForUserRequest mPassword) =
+    disableLH userLHStatus = do
       ensureReAuthorised zusr mPassword
       Client.removeLegalHoldClientFromUser uid
       LHService.removeLegalHold tid uid
       -- TODO: send event at this point (see also: related TODO in this module in
       -- 'approveDevice' and
       -- https://github.com/wireapp/wire-server/pull/802#pullrequestreview-262280386)
-      LegalHoldData.setUserLegalHoldStatus tid uid UserLegalHoldDisabled
+      changeLegalholdStatus tid uid userLHStatus UserLegalHoldDisabled
+
+changeLegalholdStatus :: TeamId -> UserId -> UserLegalHoldStatus -> UserLegalHoldStatus -> Galley ()
+changeLegalholdStatus tid uid oldLhStatus lhStatus = do
+  case (hasLH oldLhStatus, hasLH lhStatus) of
+    (False, False) -> pure ()
+    (True, True) -> pure ()
+    (False, True) -> do
+      blockConnectionsFrom1on1s
+      LegalHoldData.setUserLegalHoldStatus tid uid lhStatus
+    (True, False) -> do
+      LegalHoldData.setUserLegalHoldStatus tid uid lhStatus
+      void $ putConnectionInternal (RemoveLHBlocksInvolving uid)
+  where
+    hasLH :: UserLegalHoldStatus -> Bool
+    hasLH = \case
+      UserLegalHoldDisabled -> False
+      UserLegalHoldPending -> True
+      UserLegalHoldEnabled -> True
+      UserLegalHoldNoConsent -> False
+
+    -- FUTUREWORK: make this async
+    blockConnectionsFrom1on1s :: Galley ()
+    blockConnectionsFrom1on1s = do
+      errmsgs <-
+        iterateConversations uid (toRange (Proxy @500)) $ \convs -> do
+          conflicts <- mconcat <$> findConflicts (filter ((== One2OneConv) . cnvType) convs)
+          blockConflicts uid conflicts
+      case mconcat errmsgs of
+        [] -> pure ()
+        msgs@(_ : _) -> do
+          Log.warn $ Log.msg @String (intercalate ", " msgs)
+          throwM legalHoldCouldNotBlockConnections
+      where
+        findConflicts :: [Conversation] -> Galley [[UserId]]
+        findConflicts convs = do
+          let otherUids :: [Qualified UserId] =
+                concatMap (fmap omQualifiedId . cmOthers . cnvMembers) convs
+          ownDomain <- viewFederationDomain
+          let (_remoteUsers, localUids) = partitionRemoteOrLocalIds ownDomain otherUids
+          -- FUTUREWORK: Handle remoteUsers here when federation is implemented
+          for (chunksOf 32 localUids) $ \others -> do
+            teamsOfUsers <- Data.usersTeams others
+            filterM (shouldBlock teamsOfUsers) others
+
+        blockConflicts :: UserId -> [UserId] -> Galley [String]
+        blockConflicts _ [] = pure []
+        blockConflicts userLegalhold othersToBlock@(_ : _) = do
+          status <- putConnectionInternal (BlockForMissingLHConsent userLegalhold othersToBlock)
+          pure $ ["blocking users failed: " <> show (status, othersToBlock) | status /= status200]
+
+        chunksOf :: Int -> [any] -> [[any]]
+        chunksOf _maxSize [] = []
+        chunksOf maxSize uids = case splitAt maxSize uids of (h, t) -> h : chunksOf maxSize t
```

Comment:
What is the advantage of defining this function even though `chunksOf` from https://hoogle.zinfra.io/file/root/.stack/snapshots/x86_64-linux/e2cc9ab01ac828ffb6fe45a45d38d7ca6e672fb9fe95528498b990da673c5071/8.8.4/doc/split-0.2.3.4/Data-List-Split.html#v:chunksOf already exists in `Data.List.Split` ?

Reply:
no advandage, we just weren't looking.  fixed: da45b70af1feee8317d6ff314bed693e945e54a8